### ISSUE TITLE
`__future__` imports allow dots before them

### ISSUE THREAD

**Issue by thatbirdguythatuknownot** [python/cpython#118216]
# Bug report

### Bug description:

`__future__` imports don't check the amount of dots while checking `from`-`import` nodes for the module name, so they are still valid even if dots are put before the module name.
```pycon
>>> from .__future__ import barry_as_FLUFL
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
    from .__future__ import barry_as_FLUFL
ImportError: attempted relative import with no known parent package
>>> 1 <> 2
True
```
Is this intended? Allowing dots before the module name seems improper considering `__future__` is supposed to be part of the stdlib.

### CPython versions tested on:

3.12, 3.13

### Operating systems tested on:

Windows

### Linked PRs
* gh-118267

**Comment by terryjreedy** [comment]
What is weird is both raising an error and doing the pseudoimport anyway.

**Comment by JelleZijlstra** [comment]
Thanks @thatbirdguythatuknownot!

I merged the fix into main (3.13), so I'm closing this.

Arguably this is a bug that could also be fixed in 3.12, but I'd prefer to leave 3.12 unchanged, as the bug is rather harmless in practice and I feel core language behavior shouldn't change in a bugfix release.
